core/prob_dist.py

```diff
@@ -100,3 +100,93 @@ def test_prob_creation_and_validation(self):
 
         # check if this works
         _ = ProbabilityDist(dist)
+
+
+def get_mean_log_prob(prob_dist: ProbabilityDist, data_block) -> float:
+    """computes the average log_probability of the input data_block given the probability distribution
+    prob_dist. This acts as a fundamental limit on what any compressor designed for distribution
```

Comment:
"This acts as a fundamental limit on what any ..." - I would make this weaker statement. It's intuitively correct for most standard entropy coders "designed for distribution prob_dist", but one can always created weird codes that give small codewords to low probability symbols to beat this bound for certain sequences

Reply:
rewrote as 
```
NOTE: although not a funcdamental limit, this is roughly what a compressor designed for the given distribution prob_dist will aim to achieve"
```

Reply:
I don't see the change in the latest committed files.

Reply:
I did update it a bit, but still kept the "fundamental limit" business.. ll update

Reply:
English can be improved, but I will merge it for now. I think I will have a look at all the comments before class just to see the language makes sense. I think many of my comments are also not good grammatically.